lightning/src/routing/gossip.rs

```diff
@@ -130,6 +138,28 @@ impl Readable for NodeId {
 	}
 }
 
+impl From<PublicKey> for NodeId {
+	fn from(pubkey: PublicKey) -> Self {
+		Self::from_pubkey(&pubkey)
+	}
+}
+
+impl TryFrom<NodeId> for PublicKey {
+	type Error = secp256k1::Error;
+
+	fn try_from(node_id: NodeId) -> Result<Self, Self::Error> {
+		node_id.as_pubkey()
+	}
+}
+
+impl FromStr for NodeId {
+	type Err = bitcoin::hashes::hex::Error;
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
```

Comment:
I'm so sorry for seeing this doesn't have a newline for separation now as above. Feel free to ignore though

Reply:
Fixed, this bugs me too :sweat_smile: